internal/state/state.go

```diff
@@ -0,0 +1,43 @@
+package state
+
+import (
+	"fmt"
+	"github.com/broadinstitute/revere/internal/statuspage/statuspagetypes"
+	"sync"
+)
+
+type State struct {
+	componentNameToState *sync.Map
```

Comment:
Why does this need to be a `sync.Map` rather than a plain old `map[string]ComponentState`? As far as I can tell this data structure isn't being accessed concurrently by multiple goroutines which is the main use case for sync.Map. It would also save us from needing to perform all the typecasting that follows.

Reply:
Each pubsub message is handled in its own goroutine, and each one could try to update the status of components in parallel with the other goroutines. The docs for sync.Map said it was optimized for the "write once, read often" case ([link](https://pkg.go.dev/sync#Map)) and while I think it could be okay to use a raw map because the underlying values do per-component concurrency control, I guess I didn't want to chance it.

Reply:
sounds good, didn't realize the pub sub handler was spinning up separate go routines under the hood.

Reply:
yeah that's the whole reason for my hook thing--im super worried about "rawls goes down, instantly every alert against it fires, and now revere has to sort out the fact that rawls has 3x degraded performance, 1x partial outage, and 1x major outage, all at once"

I tried to explain this more in the comments i just pushed